packages/components/src/components/drawer/shadow.tsx

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import type { KoliBriModalEventCallbacks, LabelPropType, DrawerAPI, AlignPropType, OpenPropType, ModalPropType, DrawerStates } from '../../schema';
+import { setState, validateLabel, validateOpen, validateAlign, validateModal } from '../../schema';
+import { Component, Host, Method, Prop, State, Watch, h } from '@stencil/core';
+
+import type { JSX } from '@stencil/core';
+
+/**
+ * @slot - The Content of drawer.
+ */
+@Component({
+	tag: 'kol-drawer',
+	styleUrls: {
+		default: './style.scss',
+	},
+	shadow: true,
+})
+export class KolDrawer implements DrawerAPI {
+	public hostElement?: HTMLElement;
+	private dialogElement?: HTMLDialogElement;
+
+	@Method()
+	open() {
+		this.state = {
+			...this.state,
+			_open: true,
+		};
+		if (this.state._modal) {
+			this.dialogElement?.showModal();
+		} else {
+			this.dialogElement?.show();
+		}
+	}
+
+	@Method()
+	close() {
+		this.state = {
+			...this.state,
+			_open: false,
+		};
+		const wrapper = this.dialogElement?.querySelector('.drawer__wrapper');
+		if (!wrapper || wrapper === null) return;
+		const computedStyle = window.getComputedStyle(wrapper);
+		if (computedStyle.animationName === 'none') {
+			this.dialogElement?.close();
+			this._on?.onClose?.();
+		}
+	}
+
+	private renderDialogContent() {
+		return (
+			<div
+				class={`drawer__wrapper drawer__wrapper--${this.state._align as string} ${this.state._open ? 'drawer__wrapper--open' : `drawer__wrapper--${this.state._align as string}--close`}`}
```

Comment:
Könnten wir da noch valides BEM draus machen, also die doppelten Modifier `--...--` vermeiden?

Entweder als regulärer modifier: `drawer__wrapper--left-close`
Oder, vielleicht klarer, als state, der dann mit dem Modifier kombiniert werden kann: `drawer__wrapper--left is-closing`

(Ich fände `is-closing` auch präziser als `close`).